File: core/ui/src/main/kotlin/dev/hicka04/pokedex/core/ui/component/OfficialArtworkImage.kt
package dev.hicka04.pokedex.core.ui.component

import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import coil.compose.AsyncImage

@Composable
fun OfficialArtworkImage(url: String) {
    AsyncImage(
        model = url,
        contentDescription = "official artwork",
        modifier = Modifier
            .fillMaxWidth()
            .aspectRatio(1f)
    )
}

@Preview
@Composable
fun OfficialArtworkImagePreview() {
    OfficialArtworkImage(url = "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png")
}

File: app/ios/Core/Sources/UI/Components/OfficialArtworkImage.swift
import SwiftUI

public struct OfficialArtworkImage: View {
    let url: URL

    public init(url: URL) {
        self.url = url
    }

    public init(urlString: String) {
        self.init(url: .init(string: urlString)!)
    }

    public var body: some View {
        GeometryReader { proxy in
            AsyncImage(
                url: url,
                content: { $0.resizable() },
                placeholder: { ProgressView() }
            ).frame(
                width: min(proxy.size.width, proxy.size.height),
                height: min(proxy.size.width, proxy.size.height)
            )
        }.scaledToFit()
    }
}

#Preview("\(OfficialArtworkImage.self)") {
    OfficialArtworkImage(
        urlString: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
    )
}
